cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -820,6 +844,30 @@ protected Future<Boolean> delete(Reconciliation reconciliation) {
                 .map(Boolean.FALSE); // Return FALSE since other resources are still deleted by garbage collection
     }
 
+    /**
+     * Remove the metrics specific to the kind implementing it.
+     *
+     * @param desiredNames  Set of resources which should be reconciled
+     * @param namespace     The namespace to reconcile, or {@code *} to reconcile across all namespaces.
+     */
+    @Override
+    public void removeMetrics(Set<NamespaceAndName> desiredNames, String namespace) {
+        // Removing all CA cluster metrics not in the desired set
+        Set<String> desiredClusterNames = desiredNames.stream().map(NamespaceAndName::getName).collect(Collectors.toSet());
+        metrics.removeMetricsForCertificates(key -> {
+            if (!key.getKind().equals(kind())) {
+                return false;
+            }
+            if (namespace.equals(ConfigParameter.ANY_NAMESPACE)) {
+                // If all namespaces, remove everything for that Kind
+                return true;
+            }
```

Comment:
Why do you remove everything here? That would create race conditions when the metrics re not present. I think in your case, you should simply ignore the `namespace` parameter and rely on the `namespace` from the `desiredNames`.

Reply:
Will change it accordingly